### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by inquestruss**
Moving n2k and canbus to public projects. Many changes throughout. The readme provides a quick overview of the processing changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,57 @@
-# go-n2k
-Golang NMEA 2000 parsing library
+## boatkit-io/n2k
+
+boatkit-io/n2k comprises a package (and associated tools) that provides a subscription model enabling registered clients to receive strongly typed [go](https://go.dev) data structures generated from a stream of canbus-based NMEA 2000 messages.
```
Comment (deregtd) [inline README.md:3]: send and receive?  (we don't support it just yet but we will)

Comment (deregtd) [inline README.md:3]: NMEA 2000 messages that may come from canbus or data other sources?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,57 @@
...
+## Related Projects
+
+* [canboat](https://github.com/canboat/canboat) An open source collection of command line tools and data relevant to NMEA 2000 boat networks
+* tbd
+
+## License
+[n2k license](./LICENSE)
+
+
+
```
Comment (deregtd) [inline README.md:57]: i don't think you want to check in canboatjson.cache right?

File: cmd/pgngen/deduper.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package main
```
Comment (deregtd) [inline cmd/pgngen/deduper.go:1]: @george-e-shaw-iv what's the right way to do this?  make this an internal package -- internal/deduper/deduper.go, package deduper, and just pull it in from pgngen?  or an internal package only used for one of the cmd tools, is it right to just put it in the cmd/xxx/ dir?

File: pkg/n2k/sequence.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+type sequence struct {
+	started  time.Time
+	zero     *packet // packet 0 of sequence
+	expected uint8
+	received uint8
+	contents [MaxFrameNum + 1][]uint8 // need arrays since packets can be received out of order
+}
+
+func (s *sequence) add(p *packet) {
+	if p.frameNum == 0 {
```
Comment (deregtd) [inline pkg/n2k/sequence.go:24]: should we check in here if the frame specified has already been received and processed, and error the whole packet/reset?

File: pkg/n2k/service.go [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+	return nil
+}
+
+func (s *Service) ReplayFile(filename string) error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	file, err := os.Open(path.Join(wd, "n2kreplays", filename))
```
Comment (deregtd) [inline pkg/n2k/service.go:132]: need to make this more generic -- make it take an absolute path, and we do the n2kreplays path joining in goatkit

File: pkg/n2k/unknownpgn_test.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	u := p.unknownPGN()
+	assert.Equal(t, BAndG, u.ManufacturerCode)
+	//	assert.Equal(t, uint8(4), p.IndustryCode) Not set--not used for matches, so really don't care
+}
+
+func TestEmpty(t *testing.T) {
+	p := newPacket(can.Frame{ID: 0, Length: 0})
+	u := p.unknownPGN()
+	assert.NotEqual(t, 0, len(u.Reason.Error()))
+}
```
Comment (deregtd) [inline pkg/n2k/unknownpgn_test.go:21]: need to remove the second pkg/pgninfo_generated.go

File: README.md (hunk 1 above)
Comment (inquestruss) [reply README.md]: fixed

**Review by deregtd — APPROVED** [review]
step 1: complete!

File: pkg/n2k/sequence.go (hunk 4 above)
Comment (inquestruss) [reply pkg/n2k/sequence.go:24]: I now reset the sequence when detecting a conflict and log a warning. The conflict is same seqid and framenum from the same source, so we must have missed packets on the previous send, so we should start over. Somewhere I saw mention of a time limit for multiple frame transmission--when I find it again I'll add code to prune orphaned sequences.

File: pkg/n2k/service.go (hunk 5 above)
Comment (inquestruss) [reply pkg/n2k/service.go:132]: Deferring, since we'll move service.go out of public package.

File: pkg/n2k/unknownpgn_test.go (hunk 6 above)
Comment (inquestruss) [reply pkg/n2k/unknownpgn_test.go:21]: done.

File: cmd/pgngen/deduper.go (hunk 3 above)
Comment (inquestruss) [reply cmd/pgngen/deduper.go:1]: defer for next PR.

**Pull request merged**